keras/src/models/model_test.py

```diff
@@ -216,8 +221,8 @@ def call(self, x):
 
     def test_reviving_functional_from_config_custom_model(self):
         class CustomModel(Model):
-            def __init__(self, *args, param=1, **kwargs):
-                super().__init__(*args, **kwargs)
+            def __init__(self, inputs, outputs, *args, param=1, **kwargs):
+                super().__init__(inputs, outputs, *args, **kwargs)
```

Comment:
The carve-out for functional serialization of subclasses of Functional that have a functional-like constructor was constrained to constructors with explicit inputs and outputs arguments, which are the only ones we can test for. See implementation note 4.ii